### PR TITLE
Medical Conditions

### PR DESCRIPTION
**Pull request by Markh3tr**
betterer hazard effects.
I'll write some better desc when I get to it

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/api/data/medicalcondition/MedicalCondition.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public MedicalCondition(String name, int maxProgression, IdleProgressionType progressionType,
+                            boolean canBePermanent, Symptom.ConfiguredSymptom... symptoms) {
+        this(name, maxProgression, progressionType, 1, canBePermanent, symptoms);
+    }
+
+    public MedicalCondition(String name, int maxProgression, Symptom.ConfiguredSymptom... symptoms) {
+        this(name, maxProgression, IdleProgressionType.NONE, 0, false, symptoms);
+    }
```
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/api/data/medicalcondition/MedicalCondition.java]: Is the `idleProgressionRate` supposed to be different for these or is that an oversight?

File: src/main/java/com/gregtechceu/gtceu/api/data/medicalcondition/Symptom.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+    /**
+     * @param mobEffect MobEffect to apply
+     */
+    public Symptom(String name, int defaultStages, float defaultProgressionThreshold, Supplier<MobEffect> mobEffect) {
+        this(name, defaultStages, defaultProgressionThreshold,
+                ((hazardEffectTracker, damageSource, configuredSymptom, baseSymptom, modifier) -> hazardEffectTracker
+                        .setMobEffect(mobEffect.get(),
+                                modifier)));
+    }
```
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/api/data/medicalcondition/Symptom.java]: The formatting unfortunately makes this really hard to read.

While my suggestion may not be the best way to handle it, I'd recommend replacing the unused lambda parameters with short placeholder names to save on space, resulting in something like this:

```suggestion
    /**
     * @param mobEffect           MobEffect to apply
     * @param amplifierMultiplier amplifier added to MobEffect every progression
     */
    public Symptom(String name, int defaultStages, float defaultProgressionThreshold, MobEffect mobEffect, int amplifierMultiplier) {
        this(name, defaultStages, defaultProgressionThreshold, (tracker, $1, $2, $3, modifier) -> 
                tracker.setMobEffect(mobEffect, amplifierMultiplier * modifier)
        );
    }

    /**
     * @param mobEffect           MobEffect to apply
     * @param amplifierMultiplier amplifier added to MobEffect every progression
     */
    public Symptom(String name, int defaultStages, float defaultProgressionThreshold, Supplier<MobEffect> mobEffect, int amplifierMultiplier) {
        this(name, defaultStages, defaultProgressionThreshold, (tracker, $1, $2, $3, modifier) ->
                tracker.setMobEffect(mobEffect.get(), amplifierMultiplier * modifier)
        );
    }

    /**
     * @param mobEffect MobEffect to apply
     */
    public Symptom(String name, int defaultStages, float defaultProgressionThreshold, MobEffect mobEffect) {
        this(name, defaultStages, defaultProgressionThreshold, (tracker, $1, $2, $3, modifier) -> 
                tracker.setMobEffect(mobEffect, modifier)
        );
    }

    /**
     * @param mobEffect MobEffect to apply
     */
    public Symptom(String name, int defaultStages, float defaultProgressionThreshold, Supplier<MobEffect> mobEffect) {
        this(name, defaultStages, defaultProgressionThreshold, (tracker, $1, $2, $3, modifier) -> 
                tracker.setMobEffect(mobEffect.get(), modifier)
        );
    }
```

**Review by mikerooni — COMMENTED** [review]
I've left a suggestion for improving formatting in a hard to read place, but other than that it looks good to me so far!

File: src/main/java/com/gregtechceu/gtceu/api/data/medicalcondition/MedicalCondition.java (hunk 1 above)
Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/api/data/medicalcondition/MedicalCondition.java]: supposed to be different. it does nothing if IdleProgressionType is NONE.

**Review by TheAlaskanPhysicist — COMMENTED** [review]
LGTM

**Review by screret — APPROVED** [review]
it looked good to @stanieldev and @mikerooni so LGTM

**Review by YoungOnionMC — APPROVED** [review]

**Comment by YoungOnionMC** [comment]
looks good to merge

**Pull request merged**